modules/gdscript/gdscript_parser.cpp

```diff
@@ -5727,28 +5727,35 @@ GDScriptParser::DataType GDScriptParser::_resolve_type(const DataType &p_source,
 			}
 		}
 
-		// Still look for class constants in parent script
+		// Still look for class constants in parent scripts
 		if (!found && (base_type.kind == DataType::GDSCRIPT || base_type.kind == DataType::SCRIPT)) {
 			Ref<Script> scr = base_type.script_type;
 			ERR_FAIL_COND_V(scr.is_null(), result);
-			Map<StringName, Variant> constants;
-			scr->get_constants(&constants);
```

Comment:
I believe it's because the reference, which is the base_type, and the get_constants method only covers the constants of the script itself, which in this case is the immediately inherited script, not all the scripts which it might also inherit from, hence recursively looping to find them all.

Reply:
I probably misunderstood the intent of this function. Maybe the name should be different to make clear what it is. But it's out of the scope of this PR anyway.